cmd/keeper/keeper.go

```diff
@@ -842,6 +835,53 @@ func (p *PostgresKeeper) updateReplSlots(dbLocalState *DBLocalState, followersUI
 	return nil
 }
 
+func (p *PostgresKeeper) recreateAddonReplicationSlots(existingReplicationSlots []string, addonSlotNames []string) error {
+	// detect not stolon replication slots
+	notStolonSlots := []string{}
+	for _, currReplicationSlotName := range existingReplicationSlots {
+		if !common.IsStolonName(currReplicationSlotName) {
+			notStolonSlots = append(notStolonSlots, currReplicationSlotName)
+		}
+	}
+
+	// detect slot that should be dropped
+	slotsToDrop := []string{}
+	for _, removeCandidateSlot := range notStolonSlots {
+		if !util.StringInSlice(addonSlotNames, removeCandidateSlot) {
+			slotsToDrop = append(slotsToDrop, removeCandidateSlot)
+		}
+	}
+
+	// detect slots that should be created
+	slotsToCreate := []string{}
+	for _, addReplicationSlotName := range addonSlotNames {
+
+		if trimmedSlotName := strings.TrimSpace(addReplicationSlotName); !util.StringInSlice(notStolonSlots, trimmedSlotName) {
+			slotsToCreate = append(slotsToCreate, trimmedSlotName)
+		}
+	}
+
+	// drop unnecessary slots
+	for _, dropSlotName := range slotsToDrop {
+		log.Infow("Dropping replication slot", "dropSlotName", dropSlotName)
+		if err := p.pgm.DropReplicationSlot(dropSlotName); err != nil {
+			log.Errorw("Failed to drop replication slot", "slotName", dropSlotName, zap.Error(err))
+			return err
```

Comment:
ignore dropping errors like done in updateReplSlot since they can be in use and this will fail.